internal/dlc/refundtx.go

```diff
@@ -0,0 +1,102 @@
+package dlc
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/dgarage/dlc/internal/script"
+)
+
+// RefundTx creates refund tx
+// refund transaction
+// input:
+//   [0]:fund transaction output[0]
+//       Sequence (0xfeffffff LE)
+// output:
+//   [0]:p2wpkh a
+//   [1]:p2wpkh b
+// locktime:
+//    Value decided by contract.
+func (d *DLC) RefundTx() (*wire.MsgTx, error) {
+	tx, err := d.newRedeemTx()
+	if err != nil {
+		return nil, err
+	}
+
+	// use locktime
+	tx.TxIn[fundTxInAt].Sequence-- // max(0xffffffff-0x01)
+	tx.LockTime = d.locktime
+
+	// txouts
+	for _, p := range []Contractor{FirstParty, SecondParty} {
+		txout, err := d.refundTxOut(p)
+		if err != nil {
+			return nil, err
+		}
+		tx.AddTxOut(txout)
+	}
+
+	return tx, nil
+}
+
+func (d *DLC) refundTxOut(p Contractor) (*wire.TxOut, error) {
+	famt := d.fundAmts[p]
+	pub := d.fundTxReqs.pubs[p]
+	pkScript, err := script.P2WPKHpkScript(pub)
+	if err != nil {
+		return nil, err
+	}
+
+	txout := wire.NewTxOut(int64(famt), pkScript)
+	return txout, nil
```

Comment:
use this function
https://github.com/dgarage/dlc/blob/master/internal/dlc/dlc.go#L35